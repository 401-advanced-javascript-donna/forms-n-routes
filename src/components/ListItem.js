
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './ListItem.css';


class ListItem extends Component {

  static propTypes = {
    _id: PropTypes.string,
    name: PropTypes.string,
    photo: PropTypes.string,
    history: PropTypes.object
  }


  render() {
    const characterUrl = `/character/${this.props._id}`;
    return (
      <Link to={characterUrl}>

        <li className={styles.ListItem} key={this.props._id}>
          <img src={this.props.photo} alt={this.props._id}></img>
          <p>{this.props.name}</p>
        </li>
      </Link>
    );
  }
}

export default ListItem;
