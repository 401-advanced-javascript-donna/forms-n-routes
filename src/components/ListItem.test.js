import React from 'react';
import { shallow } from 'enzyme';
import ListItem from './ListItem';

describe('ListItem component', () => {
  it('should render ListItem', () => {
    const item = {
      _id: '5cf5679a915ecad153ab699d',
      name: 'Koh',
      photoUrl: 'https://vignette.wikia.nocookie.net/avatar/images/0/00/Koh.png/revision/latest?cb=20140124170917',
      gender: 'male'
    };
    const wrapper = shallow(<ListItem
      {...item} />);
    expect(wrapper).toMatchSnapshot();
  });
});
