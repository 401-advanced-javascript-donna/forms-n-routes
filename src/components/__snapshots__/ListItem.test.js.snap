// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component should render ListItem 1`] = `
<Link
  to="/character/5cf5679a915ecad153ab699d"
>
  <li
    className="ListItem"
    key="5cf5679a915ecad153ab699d"
  >
    <img
      alt="5cf5679a915ecad153ab699d"
    />
    <p>
      Koh
    </p>
  </li>
</Link>
`;
